src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -0,0 +1,686 @@
+import React, {
+    memo,
+    useCallback,
+    useEffect,
+    useMemo,
+    useRef,
+    useState
+} from 'react';
+import {
+    IResourcePickerProps,
+    IResourcePickerStyleProps,
+    IResourcePickerStyles
+} from './ResourcePicker.types';
+import { getStyles } from './ResourcePicker.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Spinner,
+    SpinnerSize,
+    MessageBarType,
+    MessageBar,
+    IComboBoxOption,
+    VirtualizedComboBox,
+    Icon,
+    SelectableOptionMenuItemType,
+    IRenderFunction,
+    IOnRenderComboBoxLabelProps
+} from '@fluentui/react';
+import useAdapter from '../../Models/Hooks/useAdapter';
+import {
+    AzureResourceDisplayFields,
+    AzureResourceTypes,
+    IAzureResource,
+    ValidAdtHostSuffixes,
+    ValidContainerHostSuffixes
+} from '../../Models/Constants';
+import { useTranslation } from 'react-i18next';
+import { deepCopy } from '../../Models/Services/Utils';
+
+const comboBoxOptionStyles = {
+    root: {
+        width: '100%'
+    },
+    flexContainer: { span: { width: '100%' } }
+};
+const freeformOptionsHeaderText = '---';
+const freeformOptionsHeader = {
+    key: 'additional-options',
+    text: freeformOptionsHeaderText,
+    itemType: SelectableOptionMenuItemType.Header
+};
+const getClassNames = classNamesFunction<
+    IResourcePickerStyleProps,
+    IResourcePickerStyles
+>();
+
+const ResourcePicker: React.FC<IResourcePickerProps> = ({
+    adapter,
+    resourceType,
+    requiredAccessRoles,
+    shouldFetchResourcesOnMount = true,
+    displayField = AzureResourceDisplayFields.id,
+    styles,
+    label,
+    loadingLabel,
+    searchParams,
+    onLoaded,
+    onChange,
+    additionalOptions: additionalOptionsProp,
+    selectedOption: selectedOptionProp,
+    allowFreeform = false,
+    disabled = false
+}) => {
+    const { t } = useTranslation();
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    const [options, setOptions] = useState([]);
```

Comment:
give this a type please. It's considered an any everywhere it gets used since it doesn't know the type of hte array.

Reply:
Fixed: https://github.com/microsoft/iot-cardboard-js/commit/b73eed319a2191ad298d9a707559fee259fa0fac#diff-3bd141423ed5699ed9c540576ed4472a3ed44ba1f9f6c4e73b720705561b84deR81